state/protocol/protocol_state/protocol_statemachine_test.go

```diff
@@ -13,63 +13,76 @@ import (
 	"github.com/onflow/flow-go/utils/unittest"
 )
 
-func TestUpdaterSuite(t *testing.T) {
-	suite.Run(t, new(UpdaterSuite))
+func TestProtocolStateMachine(t *testing.T) {
+	suite.Run(t, new(ProtocolStateMachineSuite))
 }
 
-// UpdaterSuite is a dedicated test suite for testing updater. It holds a minimal state to initialize updater.
-type UpdaterSuite struct {
+// BaseProtocolStateMachineSuite is a base test suite that holds common functionality for testing protocol state machines.
+// It reflects the portion of data which is present in baseProtocolStateMachine.
+type BaseProtocolStateMachineSuite struct {
 	suite.Suite
 
 	parentProtocolState *flow.RichProtocolStateEntry
 	parentBlock         *flow.Header
 	candidate           *flow.Header
-
-	updater *stateMachine
 }
 
-func (s *UpdaterSuite) SetupTest() {
+func (s *BaseProtocolStateMachineSuite) SetupTest() {
 	s.parentProtocolState = unittest.ProtocolStateFixture()
 	s.parentBlock = unittest.BlockHeaderFixture(unittest.HeaderWithView(s.parentProtocolState.CurrentEpochSetup.FirstView + 1))
 	s.candidate = unittest.BlockHeaderWithParentFixture(s.parentBlock)
+}
 
-	s.updater = newStateMachine(s.candidate.View, s.parentProtocolState)
+// ProtocolStateMachineSuite is a dedicated test suite for testing happy path state machine.
+type ProtocolStateMachineSuite struct {
+	BaseProtocolStateMachineSuite
+	stateMachine *protocolStateMachine
+}
+
+func (s *ProtocolStateMachineSuite) SetupTest() {
+	s.BaseProtocolStateMachineSuite.SetupTest()
+	var err error
+	s.stateMachine, err = newStateMachine(s.candidate.View, s.parentProtocolState)
```

Comment:
i think it is better to work with a copy here and retain `s.parentProtocolState` as a reference that `StateMachine` cannot modify

```suggestion
	s.stateMachine, err = newStateMachine(s.candidate.View, s.parentProtocolState.Copy())
```

same for other calls of `newStateMachine`:
* https://github.com/onflow/flow-go/blob/47be8dcc1e0d26799d051c16c87282de22214b09/state/protocol/protocol_state/protocol_statemachine_test.go#L68
* https://github.com/onflow/flow-go/blob/47be8dcc1e0d26799d051c16c87282de22214b09/state/protocol/protocol_state/protocol_statemachine_test.go#L83
* https://github.com/onflow/flow-go/blob/47be8dcc1e0d26799d051c16c87282de22214b09/state/protocol/protocol_state/protocol_statemachine_test.go#L95
* https://github.com/onflow/flow-go/blob/47be8dcc1e0d26799d051c16c87282de22214b09/state/protocol/protocol_state/protocol_statemachine_test.go#L104
* https://github.com/onflow/flow-go/blob/47be8dcc1e0d26799d051c16c87282de22214b09/state/protocol/protocol_state/protocol_statemachine_test.go#L138
* https://github.com/onflow/flow-go/blob/47be8dcc1e0d26799d051c16c87282de22214b09/state/protocol/protocol_state/protocol_statemachine_test.go#L157
* https://github.com/onflow/flow-go/blob/47be8dcc1e0d26799d051c16c87282de22214b09/state/protocol/protocol_state/protocol_statemachine_test.go#L167
* https://github.com/onflow/flow-go/blob/47be8dcc1e0d26799d051c16c87282de22214b09/state/protocol/protocol_state/protocol_statemachine_test.go#L189
* https://github.com/onflow/flow-go/blob/47be8dcc1e0d26799d051c16c87282de22214b09/state/protocol/protocol_state/protocol_statemachine_test.go#L208
* https://github.com/onflow/flow-go/blob/47be8dcc1e0d26799d051c16c87282de22214b09/state/protocol/protocol_state/protocol_statemachine_test.go#L230
* https://github.com/onflow/flow-go/blob/47be8dcc1e0d26799d051c16c87282de22214b09/state/protocol/protocol_state/protocol_statemachine_test.go#L264
* https://github.com/onflow/flow-go/blob/47be8dcc1e0d26799d051c16c87282de22214b09/state/protocol/protocol_state/protocol_statemachine_test.go#L323
* https://github.com/onflow/flow-go/blob/47be8dcc1e0d26799d051c16c87282de22214b09/state/protocol/protocol_state/protocol_statemachine_test.go#L238